applications/CompressiblePotentialFlowApplication/custom_processes/define_3d_wake_process.h

```diff
@@ -87,50 +127,82 @@ class KRATOS_API(COMPRESSIBLE_POTENTIAL_FLOW_APPLICATION) Define3DWakeProcess
 
     ///@}
 private:
+    ///@name Static Member Variables
+    ///@{
+        
+    /// Registry current operation
+    KRATOS_REGISTRY_ADD_PROTOTYPE("Processes.KratosMultiphysics.CompressiblePotentialFlowApplication", Process, Define3DWakeProcess)
+    KRATOS_REGISTRY_ADD_PROTOTYPE("Processes.All", Process, Define3DWakeProcess)
+    
+    ///@}
     ///@name Member Variables
     ///@{
 
-    // The airfoil model part containing the trailing edge
-    ModelPart& mrTrailingEdgeModelPart;
-    ModelPart& mrBodyModelPart;
-    ModelPart& mrStlWakeModelPart;
-    // Tolerance to avoid nodes laying exactly on the wake
-    double mTolerance;
+    // Model parts
+    ModelPart* mrTrailingEdgeModelPart = nullptr;
+    ModelPart* mrBodyModelPart = nullptr;
+    ModelPart* mrUpperSurfaceModelPart = nullptr;
+    ModelPart* mrLowerSurfaceModelPart = nullptr;
+    ModelPart* mrRootPointsModelPart = nullptr;
+    ModelPart* mrTipPointsModelPart = nullptr;
+    ModelPart* mrBluntTESurfaceModelPart = nullptr;
+    ModelPart* mrWakeModelPart = nullptr;
+    ModelPart* mrRootModelPart = nullptr;
+
+    // Tolerances 
+    double mWakeDistanceTolerance;                  // To avoid nodes laying exactly on the wake 
+    double mCheckWakeConditionTolerance;            // To check if wake conditions are fulfilled
+
     BoundedVector<double, 3> mWakeNormal;
     BoundedVector<double, 3> mWakeDirection;
     BoundedVector<double, 3> mSpanDirection;
+    BoundedVector<double, 3> mWakedrTraslation;
```

Comment:
```suggestion
    BoundedVector<double, 3> mWakeTraslationDirection;
```
To be more self explicative.